K-map minimization/k-map minimization.py

```diff
@@ -0,0 +1,453 @@
+def minFunc(numVar, stringIn):
+    '''k-map minimization'''
+    num = int(numVar)
+    x = stringIn
+    l = []
+    i = 0
+    while x[i] != 'd':
+        if x[i].isdigit() == True:  # taking the values from the input, and storing it in a list
+            if x[i + 1].isdigit() == False:
+                l.append(x[i])
+                i += 1
+            else:
+                l.append(x[i] + x[i + 1])
+                i += 2
+        else:
+            i += 1
+        y = i
+    ess = list(l)  # storing essential prime implicant in a seprate list for further use
+    if x[y + 1] != '-':
+        while x[y] != ')':
+            if x[y].isdigit() == True:
+                if x[y + 1].isdigit() == False:  # storing the dont care condition giving in the input
+                    l.append(x[y])
+                    y += 1
+                else:
+                    l.append(x[y] + x[y + 1])
+                    y += 2
+            else:
+                y += 1
+    if num == 4:
+        for i in range(len(l)):  # converting to binary for 4 variables
+            l[i] = format(int(l[i]), '04b')
+    elif num == 3:
+        for i in range(len(l)):  # converting to binary for 3 variables
+            l[i] = format(int(l[i]), '03b')
+    elif num == 2:
+        for i in range(len(l)):  # converting to binary for 2 variables
+            l[i] = format(int(l[i]), '02b')
+
+    # creating list to be stored
+    (a,b,c,d,e,f,g,h,p,q,w,r,t,y,m,n,o,z,u,v) = ([],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[])
+    l.sort()
+    count = 0
+    t = 0
+
+    #  Quine-McCluskey Method-
+
+    for i in range(len(l)):
+        for j in range(num):  # Storing the digit speratly on the basis of much many 1's they contain
+            if l[i][j] == '1':
+                count += 1
+        if count == 0:
+            a.append(l[i])
+        if count == 1:  # appending into one of the list created above for seperation
+            b.append(l[i])
+        if count == 2:
+            c.append(l[i])
+        if count == 3:
+            d.append(l[i])
+        if count == 4:
+            e.append(l[i])
+        count = 0
+
+    def step1(a, b, f):  # Proceding tp step 1 comparing a and b then follows on uptil all thrr couples are compared
```

Comment:
I think there are indentation flaws
Is this a function inside a function?
If no then indentation is bit flawed.

Reply:
Its a function inside the function. Tell me if you need any other changes.